lib/mayaUsd/ufe/UsdUndoRenameCommand.cpp

```diff
@@ -69,12 +69,16 @@ UsdUndoRenameCommand::UsdUndoRenameCommand(
     , _ufeDstItem(nullptr)
     , _stage(_ufeSrcItem->prim().GetStage())
 {
-    const UsdPrim& prim = _stage->GetPrimAtPath(_ufeSrcItem->prim().GetPath());
+    const UsdPrim prim = _stage->GetPrimAtPath(_ufeSrcItem->prim().GetPath());
 
     ufe::applyCommandRestriction(prim, "rename");
 
     // handle unique name for _newName
-    _newName = uniqueChildName(prim.GetParent(), TfMakeValidIdentifier(newName.string()));
+    const std::string validNewName = TfMakeValidIdentifier(newName.string());
+    if (validNewName != prim.GetName())
+        _newName = uniqueChildName(prim.GetParent(), validNewName);
+    else
+        _unchanged = true;
```

Comment:
Minor: instead of adding a flag, we could rely on _newName being empty.

Reply:
True, but it would make the code a little more obscure, but not by much. Since i'll move the dst assignment in the constructor, I'll also get rid of the flag.